src/main/java/org/jabref/gui/preferences/PreferencesFilterDialog.java

```diff
@@ -54,11 +61,30 @@ private void initialize() {
     }
 
     private void updateModel() {
+        auxillaryPreferenceOptions.clear();
         if (showOnlyDeviatingPreferenceOptions.isSelected()) {
-            preferenceOptions.setAll(preferencesFilter.getDeviatingPreferences());
+            auxillaryPreferenceOptions.addAll(preferencesFilter.getDeviatingPreferences());
         } else {
-            preferenceOptions.setAll(preferencesFilter.getPreferenceOptions());
+            auxillaryPreferenceOptions.addAll(preferencesFilter.getPreferenceOptions());
         }
+        preferenceOptions.setAll(auxillaryPreferenceOptions);
         count.setText(String.format("(%d)", preferenceOptions.size()));
+        String searchText = searchField.getText();
+        if (!searchText.isEmpty()) {
+            filterPreferences(searchText);
+        }
+    }
+
+    private void filterPreferences(String searchText) {
+        if (searchText.isEmpty()) {
+            updateModel();
+            return;
+        }
+
+        List<JabRefPreferencesFilter.PreferenceOption> filteredOptions = auxillaryPreferenceOptions.stream()
+                                                                                                   .filter(p -> p.getKey().toLowerCase().contains(searchText))
```

Comment:
Better use LOCALE.ROOT as second parameter for the toLowerCase function to avoid unexpected problems:
https://garygregory.wordpress.com/2015/11/03/java-lowercase-conversion-turkey/

Reply:
Understood, I have made the necessary changes.